data/mongodb/v0/tasks.go

```diff
@@ -0,0 +1,280 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/instill-ai/component/base"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type InsertInput struct {
+	Data map[string]any `json:"data"`
+}
+
+type InsertOutput struct {
+	Status string `json:"status"`
+}
+
+type FindInput struct {
+	Criteria map[string]any `json:"criteria"`
+	Limit    int            `json:"limit"`
+}
+
+type FindOutput struct {
+	Status    string           `json:"status"`
+	Documents []map[string]any `json:"documents"`
+}
+
+type UpdateInput struct {
+	Criteria map[string]any `json:"criteria"`
+	Update   map[string]any `json:"update"`
+}
+
+type UpdateOutput struct {
+	Status string `json:"status"`
+}
+
+type DeleteInput struct {
+	Criteria map[string]any `json:"criteria"`
+}
+
+type DeleteOutput struct {
+	Status string `json:"status"`
+}
+
+type DropCollectionInput struct {
+	CollectionName string `json:"collection-name"`
+}
+
+type DropCollectionOutput struct {
+	Status string `json:"status"`
+}
+
+type DropDatabaseInput struct {
+	DatabaseName string `json:"database-name"`
+}
+
+type DropDatabaseOutput struct {
+	Status string `json:"status"`
+}
+
+func (e *execution) insert(in *structpb.Struct) (*structpb.Struct, error) {
+	var inputStruct InsertInput
+	err := base.ConvertFromStructpb(in, &inputStruct)
+	if err != nil {
+		return nil, err
+	}
+
+	data := inputStruct.Data
+	ctx := context.Background()
+
+	_, err = e.collectionClient.InsertOne(ctx, data)
+	if err != nil {
+		return nil, err
+	}
+
+	outputStruct := InsertOutput{
+		Status: "Successfully inserted document",
+	}
+
+	output, err := base.ConvertToStructpb(outputStruct)
+	if err != nil {
+		return nil, err
+	}
+	return output, nil
+}
+
+func (e *execution) find(in *structpb.Struct) (*structpb.Struct, error) {
+	var inputStruct FindInput
+	err := base.ConvertFromStructpb(in, &inputStruct)
+	if err != nil {
+		return nil, err
+	}
+
+	criteria := inputStruct.Criteria
+	limit := inputStruct.Limit
+	ctx := context.Background()
+
+	realCriteria := bson.M{}
+	for key, value := range criteria {
+		if value != nil {
+			realCriteria[key] = value
+		}
+	}
+
+	findOptions := options.Find()
+
+	if limit > 0 {
+		findOptions.SetLimit(int64(limit))
+	}
+
+	var cursor *mongo.Cursor
+	if len(criteria) == 0 {
+		// Find all documents with the specified options (including limit if set)
+		cursor, err = e.collectionClient.Find(ctx, realCriteria, findOptions)
+	} else {
+		// Define the projection to include only the specified fields
+		projection := bson.M{}
+		projection["_id"] = 0
+		for key := range criteria {
+			projection[key] = 1
+		}
+		findOptions.SetProjection(projection)
+
+		// Find all documents with the specified options (including limit if set)
+		cursor, err = e.collectionClient.Find(ctx, realCriteria, findOptions)
+	}
```

Comment:
Just my preference. You can just take a look to check if they are same and neater.


```
if len(criteria) != 0 {
		projection := bson.M{}
		projection["_id"] = 0
		for key := range criteria {
			projection[key] = 1
		}
		findOptions.SetProjection(projection)
}
cursor, err = e.collectionClient.Find(ctx, realCriteria, findOptions)

```
	if len(criteria) == 0 {
		// Find all documents with the specified options (including limit if set)
		cursor, err = e.collectionClient.Find(ctx, realCriteria, findOptions)
	} else {
		// Define the projection to include only the specified fields
		projection := bson.M{}
		projection["_id"] = 0
		for key := range criteria {
			projection[key] = 1
		}
		findOptions.SetProjection(projection)

		// Find all documents with the specified options (including limit if set)
		cursor, err = e.collectionClient.Find(ctx, realCriteria, findOptions)
	}
```

Reply:
And another opinion & preference.
Code should explain itself. So, I usually do not add comments to explain what code does but why code is written like it.

e.g. [Sample I recently did](https://github.com/instill-ai/component/blob/main/operator/text/v0/markdown_splitter.go#L90)
It is not a perfect sample. I just want to share my habit.
If you think it is a good point, you can follow it. It will be your call!